File: Student List/student_list.py
import sys
from collections import OrderedDict

def main():
    students = OrderedDict()

    for i in range(1, 4):
        student_number = input(f"Enter student number for classmate {i}: ")
        first_name = input(f"Enter first name for classmate {i}: ")
        students[student_number] = first_name

    print("Current entries in the map:")
    for student_number, first_name in students.items():
        print(f"Student Number: {student_number}, First Name: {first_name}")

    third_key = list(students.keys())[2]
    del students[third_key]

    my_student_number = input("Enter your student number: ")
    my_first_name = input("Enter your first name: ")
    students[my_student_number] = my_first_name

    print("Updated entries in the map:")
    for student_number, first_name in students.items():
        print(f"Student Number: {student_number}, First Name: {first_name}")

if __name__ == "__main__":
    main()

File: Student List/StudentList.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class StudentList {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Map<String, String> students = new HashMap<>();

        for (int i = 1; i <= 3; i++) {
            System.out.println("Enter student number for classmate " + i + ":");
            String studentNumber = scanner.nextLine();
            System.out.println("Enter first name for classmate " + i + ":");
            String firstName = scanner.nextLine();
            students.put(studentNumber, firstName);
        }

        System.out.println("Current entries in the map:");
        for (Map.Entry<String, String> entry : students.entrySet()) {
            System.out.println("Student Number: " + entry.getKey() + ", First Name: " + entry.getValue());
        }

        String thirdKey = (String) students.keySet().toArray()[2];
        students.remove(thirdKey);

        System.out.println("Enter your student number:");
        String myStudentNumber = scanner.nextLine();
        System.out.println("Enter your first name:");
        String myFirstName = scanner.nextLine();
        students.put(myStudentNumber, myFirstName);

        System.out.println("Updated entries in the map:");
        for (Map.Entry<String, String> entry : students.entrySet()) {
            System.out.println("Student Number: " + entry.getKey() + ", First Name: " + entry.getValue());
        }

        scanner.close();
    }
}
